src/app/features/dashboard-widgets/entity-count-dashboard-widget/entity-count-dashboard/entity-count-dashboard.component.ts

```diff
@@ -89,29 +111,38 @@ export class EntityCountDashboardComponent
       this.entityType = "Child";
     }
 
-    const groupByType = this._entity.schema.get(this.groupBy);
-    this.groupedByEntity =
-      groupByType.dataType === EntityDatatype.dataType
-        ? groupByType.additional
-        : undefined;
-
-    const entities = await this.entityMapper.loadType(this._entity);
-    this.updateCounts(entities.filter((e) => e.isActive));
+    for (const groupByField of this.groupBy) {
+      await this.loadDataForGroupBy(groupByField);
+    }
     this.label = this._entity.labelPlural;
     this.entityIcon = this._entity.icon;
   }
 
+  async loadDataForGroupBy(groupByField: string) {
+    const entities = await this.entityMapper.loadType(this._entity);
```

Comment:
we don't have to load the entities for each grouping afresh. Let's move this outside of this function and add a method parameter to pass the same array of entities into each call of `loadDataForGroupBy(groupByField, entities)`